src/document-state.js

```diff
@@ -112,6 +116,32 @@ export class DocumentState {
   onVisibilityChanged_() {
     this.visibilityObservable_.fire();
   }
+
+  /**
+   * @param {function()} handler
```

Comment:
`setTimeout` can be pretty slow (up to event queue yield + 4ms). Promises avoid the problem (always async via a micro task). For a non-promise API `Promise.resolve().then(cb)` works.